googleapis/sample-tester
# Fix devcheck for POSIX

Also refactor for readibility and simplicity.

So, I couldn't run `devcheck` because `find -printf` is not POSIX and `-printf` is not supported.

Same goes for the `--suffix` argument to `mktemp` which doesn't work with Non-GNU `mktemp`.

Changes I made:
 - Add explicit hash bang to everything
 - Use `/bin/bash` not `/bin/sh` which is not guaranteed to be BASH and there were BASH-specific variables being used
 - Run `sample-tester` via `sample-tester`
 - Use long arguments to `unittest` for clarity
 - Expand loops and conditionals for clarity
 - Use `-v detailed` to STDOUT rather than xunit files
    - xunit output files were not being tested
    - better to see the failing output locally & in CircleCI

`tl;dr` I couldn't run these tests on my development machine before. Now I can.

### Updates:

 - Now searching for any `test-*.sh` file to execute
 - The `unittest` tests are run via a `test-*.sh` as well, so they show up alongside the other tests

**Output:**

![Screen Shot 2019-03-14 at 6 43 48 PM](https://user-images.githubusercontent.com/280/54402421-26fba280-4689-11e9-8c01-9f9cd8882557.png)
